src/node/rpc/forwarder.h

```diff
@@ -34,7 +34,13 @@ namespace ccf
 
     using ForwardedCommandId = ForwardedHeader_v2::ForwardedCommandId;
     ForwardedCommandId next_command_id = 0;
-    std::unordered_map<ForwardedCommandId, threading::TaskQueue::TimerEntry>
+
+    struct TimeoutTask {
+      threading::TaskQueue::TimerEntry timer_entry;
+      uint16_t thread_id;
+    };
+
+    std::unordered_map<ForwardedCommandId, TimeoutTask>
```

Comment:
@eddyashton This is not the way we agreed, but I think it's necessary.
Originally I tried to get the thread ID from session ID when forwarding response is received to delete the timer task from the right thread.

[b30dc0a](https://github.com/microsoft/CCF/pull/5035/commits/b30dc0a207edffe0db79058a9e8cdb6e5f5fd12b)

But it's failing in CI and the message indicates that it's still failing to delete the timer task.
I think it's because the session ID when adding timer task and the one when deleting the timer task are different.